crates/storage/db/src/lockfile.rs

```diff
@@ -64,12 +63,14 @@ impl StorageLock {
 
 impl Drop for StorageLock {
     fn drop(&mut self) {
+        // The lockfile is not created in disable-lock mode, so we don't need to delete it.
+        #[cfg(feature = "disable-lock")]
```

Comment:
Thank you so much for pointing that out. I almost introduced a bug because of a small change. I should have been more careful.

Applied this review suggestion at [a7aa559](https://github.com/paradigmxyz/reth/pull/13355/commits/a7aa559528284002c8469891c043a0222d79eaea) , PTAL